Create lowndes_fellowship_post.md (#1600)
betterscientificsoftware/bssw.io: Articles/Blog/lowndes_fellowship_post.md

# How open source tooling is changing the way professional researchers learn to code

> <!---TODO: add Hero image--->

Author: ![](images/Blog_openscapes_grassland_1500px.png)
```suggestion
![](images/Blog_openscapes_grassland_1500px.png)
```

> The same open source tooling that is changing how we do science --- R, Python, Git, GitHub --- is changing the way we teach professional researchers how to code. Tooling like Jupyter Notebooks, RMarkdown, and now Quarto enable researchers to combine narrative, code, and outputs like figures in the same document, enabling transparency and reproducibility workflows that shared openly on GitHub. This same setup can be used to teach professional researchers to code, so that the teaching narrative unfolds interspersed with code the learner can run. But what is truly remarkable is the culture shift enabled by this setup --- learners have these documents in front of them that reinforces what the teacher screenshares live, and continues to be there for reference and self-paced learning when the teacher isn't there at all. This builds trust and resilience with learners, and role-models a new way of working openly --- all of this is a paradigm shift for how professional researchers learn new skills, work, and then teach others in turn. I am a marine scientist who can say from experience that this mode of teaching is game-changing; in fact I am making it my career to amplify this to empower more researchers. For my BSSw Fellowship, I modernized Openscapes' open educational resources that I first built with RMarkdown by refactoring them in Quarto, and have since used these updated resources to teach 6 cohorts (50 teams) of professional environmental and Earth scientists at NASA, NOAA, and the EPA.

Author: ```suggestion
The same open source tooling that is changing how we do science --- R, Python, Git, GitHub --- is changing the way we teach professional researchers how to code. Tooling like Jupyter Notebooks, RMarkdown, and now Quarto enable researchers to combine narrative, code, and outputs like figures in the same document, enabling transparency and reproducibility workflows that shared openly on GitHub. This same setup can be used to teach professional researchers to code, so that the teaching narrative unfolds interspersed with code the learner can run. Since the resources are openly available, and learners can work and learn in the same tools used to create those resources, there is consistency between what they see in the materials they learn from, what they see on their end, and the expected outcomes. But what is truly remarkable is the culture shift enabled by this setup --- learners have these documents in front of them that reinforces what the teacher screenshares live, and continues to be there for reference and self-paced learning when the teacher isn't there at all. This builds trust and resilience with learners, and role-models a new way of working openly --- all of this is a paradigm shift for how professional researchers learn new skills, work, and then teach others in turn. I am a marine scientist who can say from experience that this mode of teaching is game-changing; in fact I am making it my career to amplify this to empower more researchers. For my BSSw Fellowship, I modernized Openscapes' open educational resources that I first built with RMarkdown by refactoring them in Quarto, and have since used these updated resources to teach 6 cohorts (50 teams) of professional environmental and Earth scientists at NASA, NOAA, and the EPA.
```

It was through Jenny Bryan that I first felt this trust and empowerment as a new coder and learner. Jenny Bryan, then a statistics professor at University of British Columbia and now a software engineer at Posit (formerly RStudio), shared her [Stat 545 Course](https://stat545.com/) materials as open source book (made originally with RMarkdown and now Quarto) that were a delight and light-hearted to read as it walks you through highly technical material for data wrangling, exploration, and analysis with R. Jenny went on to create additional books that are still my go-to resources and recommendations: [Happy Git With R](https://happygitwithr.com/) (which is in my opinion the best place to learn Git/GitHub, regardless of whether you are an R user), and [What They Forgot to Teach You About R](https://rstats.wtf/) (designed for an audience that "has a moderate amount of R and RStudio experience; is largely self-taught; suspects they have drifted into some idiosyncratic habits that may slow them down or make their work products more brittle; is interested in (re)designing their R lifestyle, to be more effective and more self-sufficient").

> When she teaches, Jenny provides these resources openly to her students, breaking the paradigm of the teacher having all the power and withholding information from the learners. Instead, she trusts that we we are there to learn, and having these resources in front of us reinforces that goal. And since these books are also Jenny's teaching scripts, it also has the effect of having everything she needs in one place. Being open with her script and sharing it with us in advance, we feel trusted and empowered as peers rather than infantalizing us as learners. As a learner, having the book open in my browser meant that I could reference it if I had been looking at my keyboard instead of the screen, and didn't need to interrupt the class to say "wait could you repeat that?" Sidenote: Jenny live-codes as she teaches, role-modeling that coders make typos and get errors and look things up, and further building confidence in ourselves as learners.

Author: ```suggestion
When she teaches, Jenny provides these resources openly to her students, breaking the paradigm of the teacher being the sole source of information; she is teaching live what I can also reference from a tab in my browser. As a learner, having the book open in my browser helps me because looking between my own console and the teacher's screen means I can miss syntax and now I don't fall behind or have to interrupt the class to say "could you scroll up ?". Futher, since these books are also Jenny's teaching narrative, it also has the effect of having everything she needs in one place. Being open with her script and sharing it with us in advance, we feel trusted and empowered as peers. Sidenote: Jenny live-codes as she teaches, role-modeling that coders make typos and get errors and look things up, and further building confidence in ourselves as learners.
```

> <!---TODO: add RMarkdown-Quarto images--->

Author: ```suggestion
![](images/Blog_openscapes_series_bookdown.png)
![](images/Blog_openscapes_series_quarto.png)

*Comparison of Openscapes Champions Lesson Series created originally in RMarkdown's Bookdown (left) and updated in Quarto (right). Available at https://openscapes.org/series.*
```
